### PR TITLE
feat(easy hunting): add easy hunting stats screen.

### PR DESCRIPTION
**Pull request by frenautvh**
easyhunting :
add alias -> configuration -> stats screen

### CODE REVIEW

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_number"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_queue_time"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_time"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_agent"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_action"></th>
+                    </tr>
+                </thead>
+                <tbody data-ng-if="!StatsEasyHuntingCtrl.isLoading">
+                    <tr data-ng-repeat="call in StatsEasyHuntingCtrl.callsPaginated track by call.id">
+                        <td data-title="{{:: 'telephony_alias_configuration_mode_calls_number' | translate }}">
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html]: For the `callerIdNumber` you can replace your `<td>` with a `<th scope="row">` nope?

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                <thead>
+                    <tr>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_agent"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_agent_status"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_agent_updated"></th>
+                    </tr>
+                </thead>
+                <tbody data-ng-if="!StatsEasyHuntingCtrl.isLoading">
+                    <tr data-ng-repeat="agent in StatsEasyHuntingCtrl.agentsStatusPaginated track by agent.agentId">
+                        <td data-title="{{:: 'telephony_alias_configuration_mode_calls_agent' | translate }}">
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html]: For the `agentId` you can replace your `<td>` with a `<th scope="row">` nope?

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/translations/Messages_fr_FR.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   <translation id="telephony_alias_configuration_mode_calls_action_transfert_success">Transfert de l'appel en cours</translation>
+   <translation id="telephony_alias_configuration_mode_calls_action_eavesdrop">Écouter ou chuchoter</translation>
+   <translation id="telephony_alias_configuration_mode_calls_action_eavesdrop_action">Action</translation>
+   <translation id="telephony_alias_configuration_mode_calls_action_eavesdrop_type_listen">Écouter</translation>
+   <translation id="telephony_alias_configuration_mode_calls_action_eavesdrop_type_whisper_agent">Écouter et chuchoter à l'agent</translation>
+   <translation id="telephony_alias_configuration_mode_calls_action_eavesdrop_type_whisper_all">Écouter et chuchoter à tous</translation>
+   <translation id="telephony_alias_configuration_mode_calls_action_eavesdrop_number">Numéro</translation>
+   <translation id="telephony_alias_configuration_mode_calls_action_eavesdrop_help">Vous pouvez écouter une communication en cours ou de chuchoter dans celle-ci. Pour cela, il vous suffit de renseigner un numéro de téléphone qui sera appelé automatiquement et mis en relation avec la communication en cours.</translation>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/translations/Messages_fr_FR.xml]: `ou de chuchoter` => `ou chuchoter` (-`de`)

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/eavesdrop/telecom-telephony-alias-configuration-stats-easyHunting-eavesdrop.controller.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+angular.module("managerApp").controller("TelecomTelephonyAliasConfigurationStatsEasyHuntingEavesdropCtrl", function ($scope, $uibModalInstance, $translate, Telephony, Toast, params) {
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/eavesdrop/telecom-telephony-alias-configuration-stats-easyHunting-eavesdrop.controller.js]: `$scope` is not used in your controller.

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/hangup/telecom-telephony-alias-configuration-stats-easyHunting-hangup.controller.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+angular.module("managerApp").controller("TelecomTelephonyAliasConfigurationStatsEasyHuntingHangupCtrl", function ($scope, $uibModalInstance, $translate, Telephony, Toast, params) {
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/hangup/telecom-telephony-alias-configuration-stats-easyHunting-hangup.controller.js]: `$scope` is not used in your controller.

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/intercept/telecom-telephony-alias-configuration-stats-easyHunting-intercept.controller.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+angular.module("managerApp").controller("TelecomTelephonyAliasConfigurationStatsEasyHuntingInterceptCtrl", function ($scope, $uibModalInstance, $translate, Telephony, Toast, params) {
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/intercept/telecom-telephony-alias-configuration-stats-easyHunting-intercept.controller.js]: `$scope` is not used in your controller.

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/transfert/telecom-telephony-alias-configuration-stats-easyHunting-transfert.controller.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+angular.module("managerApp").controller("TelecomTelephonyAliasConfigurationStatsEasyHuntingTransfertCtrl", function ($scope, $uibModalInstance, $translate, Telephony, Toast, params) {
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/transfert/telecom-telephony-alias-configuration-stats-easyHunting-transfert.controller.js]: `$scope` is not used in your controller.

**Comment by KamiAmiga** [comment]
@antleblanc : indeed. I prefer using <span> for some semantic purposes, but as most of the icons have aria-hidden set to true, it may not be an obligation.

**Comment by frisouilles** [comment]
@Jisay

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+                </thead>
+                <tbody data-ng-if="!StatsEasyHuntingCtrl.isLoading">
+                    <tr data-ng-repeat="call in StatsEasyHuntingCtrl.callsPaginated track by call.id">
+                        <th scope="row" data-title="{{:: 'telephony_alias_configuration_mode_calls_number' | translate }}">
+                            <span class="text-nowrap right-space-m16 text-capitalize"
+                                  data-ng-bind="call.callerIdNumber | phoneNumber"></span>
+                            <span class="text-nowrap right-space-m16"
+                                  data-ng-if="call.callerIdName"
+                                  data-ng-bind="'(' + call.callerIdName + ')'"></span>
+                        </td>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html]: Should be `</th>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_agent"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_agent_status"></th>
+                        <th scope="col" data-translate="telephony_alias_configuration_mode_calls_agent_updated"></th>
+                    </tr>
+                </thead>
+                <tbody data-ng-if="!StatsEasyHuntingCtrl.isLoading">
+                    <tr data-ng-repeat="agent in StatsEasyHuntingCtrl.agentsStatusPaginated track by agent.agentId">
+                        <th scope="row" data-title="{{:: 'telephony_alias_configuration_mode_calls_agent' | translate }}">
+                            <span class="text-nowrap right-space-m16" data-ng-bind="agent.agentId"></span>
+                        </td>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html]: Should be `</th>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+                        <td data-title="{{:: 'telephony_alias_configuration_mode_calls_agent_updated' | translate }}">
+                            <span data-translate="telephony_alias_configuration_mode_calls_agent_updated_ago"></span>
+                            <elapsed-time data-from="agent.lastStatusChange"></elapsed-time>
+                        </td>
+                    </tr>
+                </tbody>
+                <tbody data-ng-if="StatsEasyHuntingCtrl.isLoading">
+                    <tr>
+                        <td class="text-center top-space-p16 bottom-space-p16" colspan="4">
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html]: `colspan="4"` => `colspan="3"` nope?

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+                            <spinner></spinner>
+                        </td>
+                    </tr>
+                </tbody>
+                <tbody data-ng-if="!StatsEasyHuntingCtrl.isLoading && !StatsEasyHuntingCtrl.agentsStatus.length">
+                    <tr>
+                        <td class="text-center top-space-p16 bottom-space-p16" colspan="4">
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/telecom-telephony-alias-configuration-stats-easyHunting.html]: `colspan="4"` => `colspan="3"` nope?

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/eavesdrop/telecom-telephony-alias-configuration-stats-easyHunting-eavesdrop.html [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+        <form name="eavesdropForm"
+              data-ng-submit="$ctrl.submit()"
+              novalidate>
+            <div class="form-group">
+                <label for="type"
+                       class="control-label"
+                       data-translate="telephony_alias_configuration_mode_calls_action_eavesdrop_action">
+                </label>
+                <select id="type"
+                        name="socialReason"
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/eavesdrop/telecom-telephony-alias-configuration-stats-easyHunting-eavesdrop.html]: `socialReason` => `type`?

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/eavesdrop/telecom-telephony-alias-configuration-stats-easyHunting-eavesdrop.html [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+                       class="form-control"
+                       data-ng-model="$ctrl.number"
+                       data-ng-pattern="/^\+?\d+$/"
+                       autofocus
+                       required />
+                <p class="alert alert-danger top-space-m16"
+                   data-ng-if="$ctrl.error"
+                   role="alert">
+                    <i class="ovh-font ovh-font-filled-error" aria-hidden="true"></i>
+                    <span data-ng-bind="$ctrl.error"></p>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/eavesdrop/telecom-telephony-alias-configuration-stats-easyHunting-eavesdrop.html]: `</p>` => `</span>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/hangup/telecom-telephony-alias-configuration-stats-easyHunting-hangup.html [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+                aria-label="{{ ::'cancel' | translate }}"
+                data-ng-click="$ctrl.cancel()">
+            <span aria-hidden="true">&times;</span>
+        </button>
+    </div>
+
+    <div class="modal-body">
+        <p class="alert alert-info" role="alert">
+            <i class="ovh-font ovh-font-info" aria-hidden="true"></i>
+            <span data-translate="telephony_alias_configuration_mode_calls_action_hangup_help"></p>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/hangup/telecom-telephony-alias-configuration-stats-easyHunting-hangup.html]: `</p>` => `</span>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/hangup/telecom-telephony-alias-configuration-stats-easyHunting-hangup.html [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+        </p>
+        <form name="hangupForm"
+              data-ng-submit="$ctrl.submit()"
+              novalidate>
+            <div class="form-group">
+                <p class="alert alert-danger top-space-m16"
+                   data-ng-if="$ctrl.error"
+                   role="alert">
+                    <i class="ovh-font ovh-font-filled-error" aria-hidden="true"></i>
+                    <span data-ng-bind="$ctrl.error"></p>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/hangup/telecom-telephony-alias-configuration-stats-easyHunting-hangup.html]: `</p>` => `</span>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/intercept/telecom-telephony-alias-configuration-stats-easyHunting-intercept.html [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+                aria-label="{{ ::'cancel' | translate }}"
+                data-ng-click="$ctrl.cancel()">
+            <span aria-hidden="true">&times;</span>
+        </button>
+    </div>
+
+    <div class="modal-body">
+        <p class="alert alert-info" role="alert">
+            <i class="ovh-font ovh-font-info" aria-hidden="true"></i>
+            <span data-translate="telephony_alias_configuration_mode_calls_action_intercept_help"></p>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/intercept/telecom-telephony-alias-configuration-stats-easyHunting-intercept.html]: `</p>` => `</span>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/intercept/telecom-telephony-alias-configuration-stats-easyHunting-intercept.html [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+                       class="form-control"
+                       data-ng-model="$ctrl.number"
+                       data-ng-pattern="/^\+?\d+$/"
+                       autofocus
+                       required />
+                <p class="alert alert-danger top-space-m16"
+                   data-ng-if="$ctrl.error"
+                   role="alert">
+                    <i class="ovh-font ovh-font-filled-error" aria-hidden="true"></i>
+                    <span data-ng-bind="$ctrl.error"></p>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/intercept/telecom-telephony-alias-configuration-stats-easyHunting-intercept.html]: `</p>` => `</span>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/transfert/telecom-telephony-alias-configuration-stats-easyHunting-transfert.html [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+                aria-label="{{ ::'cancel' | translate }}"
+                data-ng-click="$ctrl.cancel()">
+            <span aria-hidden="true">&times;</span>
+        </button>
+    </div>
+
+    <div class="modal-body">
+        <p class="alert alert-info" role="alert">
+            <i class="ovh-font ovh-font-info" aria-hidden="true"></i>
+            <span data-translate="telephony_alias_configuration_mode_calls_action_transfert_help"></p>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/transfert/telecom-telephony-alias-configuration-stats-easyHunting-transfert.html]: `</p>` => `</span>`

File: client/app/telecom/telephony/alias/configuration/stats/easyHunting/transfert/telecom-telephony-alias-configuration-stats-easyHunting-transfert.html [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+                       class="form-control"
+                       data-ng-model="$ctrl.number"
+                       data-ng-pattern="/^\+?\d+$/"
+                       autofocus
+                       required />
+                <p class="alert alert-danger top-space-m16"
+                   data-ng-if="$ctrl.error"
+                   role="alert">
+                    <i class="ovh-font ovh-font-filled-error" aria-hidden="true"></i>
+                    <span data-ng-bind="$ctrl.error"></p>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/stats/easyHunting/transfert/telecom-telephony-alias-configuration-stats-easyHunting-transfert.html]: `</p>` => `</span>`

**Comment by Jisay** [comment]
Should also be done for ovhPabx number types (CCS expert)

**Pull request merged**